src/clinical/clinical-service.ts

```diff
@@ -420,13 +421,9 @@ export const getValidRecordsPostSubmission = async (
 
 		const schemaName = await dictionaryManager.instance().getCurrentName();
 
-		const migrationDictionary = await dictionaryManager
-			.instance()
-			.loadSchemaByVersion(schemaName, migrationVersion)
-			.catch(async (err) => {
-				L.error('getValidRecordsPostSubmission error finding migration schema', err);
-				return await dictionaryManager.instance().getCurrent();
-			});
+		const migrationDictionary =
+			(await schemaRepo.get(schemaName, undefined, migrationVersion)) ||
+			(await dictionaryManager.instance().getCurrent());
```

Comment:
A) schemaRepo.get can potentially return undefined
B) There is no 'get schema from db' function on dictionaryManager, only 'fetch schema' or 'load from rest client and save'

`getValidRecordsPostSubmission` requires a Dictionary to validate against
So if dictionary is not found in DB, fetch to Lectern becomes necessary

Reply:
ya dictionaryManager can be considered to be the LecternClient, it's meant for fetching from the remote service not for reading from the local db.